thunder/executors/torch_compile.py

```diff
@@ -84,7 +85,12 @@ def torch_interpreted_func(*args):
     # torch.compile executor"
     torch_trace = trace(inline_trace=False)(torch_interpreted_func, *sorted_unique_inputs)
     trace_callable = torch_trace.python_callable(include_decorators=False)
-    compiled_func = torch.compile(trace_callable, fullgraph=True)
+    torch_compile_fullgraph: None | bool = get_compile_option(
+        "torch_compile_fullgraph", "Whether to enable `fullgraph` from `torch.compile`. Defaults to `True`."
+    )
+    if torch_compile_fullgraph is None:
+        torch_compile_fullgraph = True
```

Comment:
I would default it to False because the effect of setting this to True is just that some programs may throw an error instead of executing, but I think the historic value is True so this just preserves it?

Reply:
I do think it is better to default it to False, however, I am not sure if there were any specific reasons on why we kept this as True in the first place.

Reply:
The hypothesis of the executor and the default to True is to only send things to torch compile that do not cause graph breaks and if we do, we want to investigate and improve the fusion logic.

Reply:
I don't know if causing a graph break is an issue or not, but passing `True` here doesn't seem to hurt anything

Reply:
That said, we might add a
```
try:
...
except torch._dynamo.exc.Unsupported:
    print("""The torch compile executor got a segment that it cannot process without graph breaks, 
you can enable graph breaks by passing `torch_compile_fullgraph=False` to thunder.jit""")
    raise
```